### PR TITLE
Ng 25 segment display

### PR DESCRIPTION
**Pull request by cfboy**
Added Seven Segment Display. (NG-25)

### CODE REVIEW

File: GUI/window.py [hunk 1]
```diff
@@ -711,6 +810,114 @@ def change_color(self, binary):
...
+
+    rightA = ListProperty([.41, .41, .41])
+    rightB = ListProperty([.41, .41, .41])
+    rightC = ListProperty([.41, .41, .41])
+    rightD = ListProperty([.41, .41, .41])
+    rightE = ListProperty([.41, .41, .41])
+    rightF = ListProperty([.41, .41, .41])
+    rightG = ListProperty([.41, .41, .41])
+
+    def __init__(self, **kwargs):
```
Comment (ChristianPerez34) [inline GUI/window.py]: Not being used aside of calling super class. Should remove

File: GUI/window.py [hunk 2]
```diff
@@ -711,6 +810,114 @@ def change_color(self, binary):
...
+        super(SevenSegmentDisplay, self).__init__(**kwargs)
+
+    # Iterates through the binary at the Input location (RAM) to determine which are 1s and which are 0s
+    # Then, activate segments accordingly.
+    def activate_segments(self, binary):
+        print(binary)
```
Comment (ChristianPerez34) [inline GUI/window.py]: Print statement not needed

File: GUI/window.py [hunk 3]
```diff
@@ -711,6 +810,114 @@ def change_color(self, binary):
...
+        control_bit = int(binary[len(binary) - 1])
```
Comment (ChristianPerez34) [inline GUI/window.py]: control_bit = int(binary[-1])

File: GUI/window.py [hunk 4]
```diff
@@ -711,6 +810,114 @@ def change_color(self, binary):
...
+        for bit in range(len(binary) - 1):
+            if control_bit == 0:
+                # if control_bit == 1 then activate the seven left segments depending of the bit.
+                if bit == 0:
+                    if binary[bit] == '0':
+                        self.leftA = (.41, .41, .41)
+                        print(bit)
```
Comment (ChristianPerez34) [inline GUI/window.py]: Remove print

File: GUI/window.py [hunk 5]
```diff
@@ -711,6 +810,114 @@ def change_color(self, binary):
...
+                        self.rightF = (.41, .41, .41)
+                    else:
+                        self.rightF = (1, 0, 0)
+                elif bit == 6:
+                    if binary[bit] == '0':
+                        self.rightG = (.41, .41, .41)
+                    else:
+                        self.rightG = (1, 0, 0)
+
+    def __init__(self, **kwargs):
```
Comment (ChristianPerez34) [inline GUI/window.py]: Not doing anything other than calling super class. Remove this method

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

**Review by ChristianPerez34 — APPROVED** [review]
You have an import that is not being used `secrets`.

**Review by alejoreyes96 — APPROVED** [review]

**Review by noelvalentin — APPROVED** [review]

**Pull request merged**